### PR TITLE
[pigeon] Add Swift Package Manager integration to example app

### PR DESCRIPTION
**Pull request by loic-sharma**
This migrates Pigeon's example app to have Swift Package Manager integration. This raises the example apps' Flutter SDK requirement to 3.24.0 or higher.

This also checks-in CocoaPods integration stuff that was missing.

I will get a text exemption for this change.

The example apps were migrated using the following commands:

```
dart run script/tool/bin/flutter_plugin_tools.dart build-examples --ios --swift-package-manager
dart run script/tool/bin/flutter_plugin_tools.dart build-examples --macos --swift-package-manager
```

Part of: https://github.com/flutter/flutter/issues/159173

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/pigeon/example/app/ios/Flutter/AppFrameworkInfo.plist [hunk 1]
```diff
@@ -21,6 +21,6 @@
   <key>CFBundleVersion</key>
   <string>1.0</string>
   <key>MinimumOSVersion</key>
-  <string>11.0</string>
+  <string>12.0</string>
```
Comment (loic-sharma) [inline packages/pigeon/example/app/ios/Flutter/AppFrameworkInfo.plist:24]: Camera equivalent:

https://github.com/flutter/packages/blob/264d920a4ce617436eeee46a8f5fa35d971f39c1/packages/camera/camera/example/ios/Flutter/AppFrameworkInfo.plist#L27-L28

File: packages/pigeon/example/app/ios/Podfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Uncomment this line to define a global platform for your project
-# platform :ios, '11.0'
+# platform :ios, '12.0'
```
Comment (loic-sharma) [inline packages/pigeon/example/app/ios/Podfile:2]: Camera equivalent:

https://github.com/flutter/packages/blob/d34f32dfaa6ac2e0af22208e1d1116d38ba47235/packages/camera/camera/example/ios/Podfile#L2

File: packages/pigeon/example/app/ios/Runner.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -10,7 +10,9 @@
 		1498D2341E8E89220040F4C2 /* GeneratedPluginRegistrant.m in Sources */ = {isa = PBXBuildFile; fileRef = 1498D2331E8E89220040F4C2 /* GeneratedPluginRegistrant.m */; };
 		3368472729F02D040090029A /* Messages.g.swift in Sources */ = {isa = PBXBuildFile; fileRef = 3368472629F02D040090029A /* Messages.g.swift */; };
 		3B3967161E833CAA004F5970 /* AppFrameworkInfo.plist in Resources */ = {isa = PBXBuildFile; fileRef = 3B3967151E833CAA004F5970 /* AppFrameworkInfo.plist */; };
+		3EE8794C275F32088AD591EB /* Pods_Runner.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = A650BDD6F68BD8FFBF3F1780 /* Pods_Runner.framework */; };
```
Comment (loic-sharma) [inline packages/pigeon/example/app/ios/Runner.xcodeproj/project.pbxproj:13]: local_auth_darwin equivalent:

https://github.com/flutter/packages/blob/264d920a4ce617436eeee46a8f5fa35d971f39c1/packages/local_auth/local_auth_darwin/example/macos/Runner.xcodeproj/project.pbxproj#L23-L33

File: packages/pigeon/example/app/ios/Runner.xcworkspace/contents.xcworkspacedata [hunk 4]
```diff
@@ -4,4 +4,7 @@
    <FileRef
       location = "group:Runner.xcodeproj">
    </FileRef>
+   <FileRef
+      location = "group:Pods/Pods.xcodeproj">
+   </FileRef>
```
Comment (loic-sharma) [inline packages/pigeon/example/app/ios/Runner.xcworkspace/contents.xcworkspacedata:9]: local_auth equivalent:

https://github.com/flutter/packages/blob/264d920a4ce617436eeee46a8f5fa35d971f39c1/packages/local_auth/local_auth_darwin/example/macos/Runner.xcworkspace/contents.xcworkspacedata#L7-L9

**Review by tarrinneal — APPROVED** [review]

**Pull request merged**